typeset/converter.js

```diff
@@ -94,26 +109,21 @@ const createMapOfMathMLElements = async (log, inputPath, cssPath, outputPath, ou
     })
   }
 
-  const mathEntries = await page.evaluate((PROGRESS_TIME) => {
+  const mathEntries = await page.evaluate((PROGRESS_TIME, PLACE_HOLDER_TEMPLATE) => {
     const mathMLNodes = document.getElementsByTagNameNS('http://www.w3.org/1998/Math/MathML', 'math')
     const latexNodes = document.querySelectorAll('[data-math]')
     console.log(`Found ${mathMLNodes.length} MathML elements and ${latexNodes.length} LaTeX functions`)
     console.info('Extracting MathML and LaTeX elements from the document...')
     const mathNodes = [...mathMLNodes, ...latexNodes]
     const total = mathNodes.length
-    const mathMap/*: Map<string, {xml: string, fontSize: number}> */ = new Map()
+    const mathMap /* [{xml: string, fontSize: number}, ...] */ = []
     let prevTime = Date.now()
     let index = 0
     for (const mathNode of mathNodes) {
       const xml = mathNode.getAttribute('data-math') ? mathNode.getAttribute('data-math') : mathNode.outerHTML
       const fontSize = parseFloat(window.getComputedStyle(mathNode, null).getPropertyValue('font-size'))
-      // only set an ID if one does not already exist
-      if (!mathNode.getAttribute('id')) {
-        mathNode.setAttribute('id', `mjnode-${index}`)
-        mathNode.classList.add('-remove-id-later')
-      }
-
-      const id = mathNode.getAttribute('id')
+      // put html comment placeholder for where the converted math should go
+      mathNode.outerHTML = PLACE_HOLDER_TEMPLATE.replace('{id}', index)
```

Comment:
Since we are iterating though `mathNodes` and it is constructed via `[...mathMLNodes, ...latexNodes]`, it is not guaranteed to be in the markups DFS order. This has implications in serialization since a latex node could appear _before_ a mathml node yet have an id _greater_ than the mathml node. If we assume that latex and mathml are never intermixed in our content, this is fine. This could probably be fixed by constructing `mathNodes` with a `querySelector` that covers both mathML and latex if possible.
I'll approve since there is now there error checking when an id cannot be found, but we should be wary of how mathify will handle content mixing MathML and LaTeX going forward.

Reply:
@tomjw64 This is now fixed in the latest commit.